src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -58,4 +62,32 @@ def get(self, svc_id):
 
         daemon['rgw_metadata'] = metadata
         daemon['rgw_status'] = status
+
         return daemon
+
+
+@ApiController('rgw/proxy')
+@AuthRequired()
+class RgwProxy(BaseController):
+    @cherrypy.expose
+    def default(self, *vpath, **params):
+        try:
+            rgw_client = RgwClient.admin_instance()
+
+        except NoCredentialsException as e:
+            cherrypy.response.headers['Content-Type'] = 'application/json'
+            cherrypy.response.status = 401
+            return json.dumps({'message': str(e)})
```

Comment:
This line triggers an exception in cherrypy when running in python 3 because of the return type. Please change this line to:
`return json.dumps({'message': str(e)}).encode('utf-8')`